drivers/volume/portworx/schedops/k8s-schedops.go

```diff
@@ -328,33 +334,32 @@ func (k *k8sSchedOps) ValidateVolumeCleanup(d node.Driver) error {
 			if found {
 				continue
 			}
-			orphanPods = append(orphanPods, podUID)
 
-			// Check if there are files under portworx volume
-			// We use a depth of 2 because the files stored in the volume are in the pvc
-			// directory under the portworx-volume folder for that pod. For instance,
-			// ../kubernetes-io~portworx-volume/pvc-<id>/<all_user_files>
 			n := nodeMap[nodeName]
-			findFileOpts := node.FindOpts{
-				ConnectionOpts: connOpts,
-				MinDepth:       2,
-				MaxDepth:       2,
-			}
-			files, _ := d.FindFiles(path, n, findFileOpts)
-			if len(strings.TrimSpace(files)) > 0 {
-				dirtyVolPods = append(dirtyVolPods, podUID)
+			// Check if /var/lib/kubelet/pods/{podUID}/volumes/kubernetes.io~portworx-volume is empty
+			if isDirEmpty(path, n, d) {
```

Comment:
shouldn't this be `!isDirEmpty()`?

Reply:
Hey @piyush-nimbalkar  as per my understanding of the discussion we had on the channel, if this dir is empty it is orphan otherwise it is dirty, does it make sense?

Reply:
That's correct. If the portworx-volume dir is empty, then why would we check the contents inside it.
If the dir is not empty, then we go and check if there is a pvc directory or not to decide it is dirty or just orphan, right?

Reply:
Sorry, now I’ve got what you mean, fixing